cidc_api/models/models.py

```diff
@@ -501,7 +564,13 @@ def insert(self, session: Session, commit: bool = True, compute_etag: bool = Tru
         try:
             # Grant IAM permission in GCS only if db insert worked
             grant_download_access(grantee.email, self.trial_id, self.upload_type)
+            # Remove permissions staged for deletion, if any
+            for perm in perms_to_delete:
+                revoke_download_access(grantee.email, perm.trial_id, perm.upload_type)
         except Exception as e:
+            # Add back deleted permissions, if any
+            for perm in perms_to_delete:
+                perm.insert()
             # Delete the just-created permissions record
             super().delete()
```

Comment:
don't you want to just roll back current session / transaction?

Reply:
The transaction has already been committed here https://github.com/CIMAC-CIDC/cidc-api-gae/pull/454/files#diff-695d07bda541617439f1fc301a4ea5b1794d003d385e0fd1361d920e060f7ab6R558, so I don't think it can be rolled back.

Some context for this: we commit the permission record insert to ensure we catch foreign key / uniqueness / etc. issues before creating a corresponding IAM permission.